runner-plugin-sdk/src/main/java/org/apache/apisix/plugin/runner/PostResponse.java

```diff
@@ -116,7 +130,27 @@ public void setHeader(String headerKey, String headerValue) {
         if (Objects.isNull(headers)) {
             headers = new HashMap<>();
         }
-        headers.put(headerKey, headerValue);
+
+        headers.put(headerKey, new ArrayList<>());
+        headers.get(headerKey).add(headerValue);
+    }
+
+    public void addHeader(String headerKey, String headerValue) {
```

Comment:
My suggestion here is to use the correct modifier type. If the method is only used in this class, it will be better to use private modification.

Reply:
made it private